compatibility/spec.yaml

```diff
@@ -19,12 +19,4 @@
 - kafka: "0.10.1.1"
   scala: ["2.11", "2.10"]
 - kafka: "0.10.1.0"
-  scala: ["2.11", "2.10"]
-- kafka: "0.10.0.1"
-  scala: ["2.11", "2.10"]
-- kafka: "0.10.0.0"
-  scala: ["2.11", "2.10"]
-- kafka: "0.9.0.1"
-  scala: ["2.11", "2.10"]
-- kafka: "0.9.0.0"
-  scala: ["2.11", "2.10"]
+  scala: ["2.11", "2.10"]
```

Comment:
What do you think of adding version `2.4.1` and `2.5.1` ? Also updating the README section with updated list.